### PR TITLE
Add server-side key-value cache

### PR DESCRIPTION
**Pull request by milenvoutchev**
Added cacheService with get, set API for the backend;
Refactored to utilize service layer.
Moved cache routes to own file.

### CODE REVIEW

File: server/src/index.js [hunk 1]
```diff
@@ -20,12 +21,14 @@ app.use(bodyParser.urlencoded({ extended: false }));
...
 
+app.locals.cacheService = new CacheService(config.defaultCacheTTL);
+
 // cache management endpoints
 app.get(`${ROUTE_PREFIX}/cache/index`, (req, res) => res.json(apicache.getIndex()));
 app.get(`${ROUTE_PREFIX}/cache/clear`, (req, res) => res.json(apicache.clear()));
 
 // Cache all routes. Note: cache could also be set-up per-route - see docs
-app.use(apicache.middleware(config.globalEndpointCache));
+// app.use(apicache.middleware(config.globalEndpointCache));  // TODO: Remove when development of cache is done
```
Comment (artemstepanenko) [inline server/src/index.js]: The comment after `TODO:` says, this line should be removed.

File: server/src/routes/newRelicErrors.js [hunk 2]
```diff
@@ -26,21 +27,27 @@ const router = express.Router();
...
+      if (err) {
+        next(err);
+        return;
+      }
+      const payload = {
+        previous: _get(insightsResponse, 'previous.results[0].count'),
+        current: _get(insightsResponse, 'current.results[0].count'),
+      };
+      res.json(getResponseSuccess(payload));
+      cacheService.set(cacheKey, payload);
```
Comment (artemstepanenko) [inline server/src/routes/newRelicErrors.js]: I'm not sure how much important it here. I think the positions of the last two lines should be switched.

Comment (milenvoutchev) [inline server/src/routes/newRelicErrors.js]: I checked and it doesn't matter, cache is saved either way.

Comment (milenvoutchev) [inline server/src/routes/newRelicErrors.js]: @artemstepanenko

**Pull request merged**